### PR TITLE
SNC: Dusk Mangler and support

### PR DESCRIPTION
**Pull request by Northmoc**
What do you think of using `_` to separate the alternative costs? I thought it might be nice to avoid `:` as it could be used for other things. `;` is out as it is used so much in cost strings.

This seems to work well for the new card as well as the old ones. Smoothed out a few things in keyword to text like unneeded "pay" always tacked on to second cost.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/dusk_mangler.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Dusk Mangler
+ManaCost:5 B B
+Types:Creature Horror
+PT:5/4
+K:AlternateAdditionalCost:Sac<1/Creature>_Discard<1/Card>_PayLife<4>
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigSac | TriggerDescription$ When CARDNAME enters the battlefield, each opponent sacrifices a creature, discards a card, and loses 4 life.
+SVar:TrigSac:DB$ Sacrifice | Defined$ Opponent | SacValid$ Creature | SubAbility$ DBDiscard
+SVar:DBDiscard:DB$ Discard | Mode$ TgtChoose | Defined$ Opponent | RememberDiscarded$ True | SubAbility$ DBLoseLife
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/dusk_mangler.txt:8]: Remove RememberDiscarded$

**Comment by Agetian** [comment]
Personally I think it's fine even if it does look slightly uglier-ish :) I can't think of any alternative option that would work though, and ":" might not work in that context as well for more complex cards, so yeah, unless someone else can suggest a cleaner approach for it, I'd say this one is fine :)

**Review by Agetian — APPROVED** [review]

**Comment by Northmoc** [comment]
> Personally I think it's fine even if it does look slightly uglier-ish :) I can't think of any alternative option that would work though, and ":" might not work in that context as well for more complex cards, so yeah, unless someone else can suggest a cleaner approach for it, I'd say this one is fine :)

You know, we could use ` | ` 🤦‍♂️

**Comment by Agetian** [comment]
If "|" is possible in that context, I'd say it looks cleaner because it looks closer to how things are divided elsewhere in the script, yeah :)

**Comment by Hanmac** [comment]
Imo ":" should have worked as separate between the costs

**Comment by Northmoc** [comment]
> Imo ":" should have worked as separate between the costs

Sure, I can make that work... hopefully this keyword wouldn't need any other splits in the future beyond the inevitable four, five, etc. alternate additional costs!

**Comment by Hanmac** [comment]
> > Imo ":" should have worked as separate between the costs
> 
> Sure, I can make that work... hopefully this keyword wouldn't need any other splits in the future beyond the inevitable four, five, etc. alternate additional costs!

just start the for loop with `i = 1` ?

**Pull request merged**